restaurant/views.py

```diff
@@ -0,0 +1,167 @@
+
+    return Response(data=serializer.data, status=status.HTTP_200_OK)
+  except (ValidationError, ObjectDoesNotExist):
+    return Response({
+        'error': 'Invalid restaurant id'
+    }, status=status.HTTP_400_BAD_REQUEST)
+
+
+@api_view(['POST'])
+@permission_classes([IsAuthenticated])
+def create_ticket(request, restaurant_id):
+  user = get_profile_from_api_request(request=request)
+
+  try:
+    restaurant = Restaurant.objects.get(id=restaurant_id, owner__id=user.id)
+    data = request.data
+    data.update({'restaurant': restaurant.id})
+    serializer = TicketSerializer(data=data)
+    serializer.is_valid(raise_exception=True)
+    serializer.save()
+
+    return Response(data=serializer.data, status=status.HTTP_201_CREATED)
+  except (ObjectDoesNotExist, ValidationError):
+    return Response({
+        'error': 'You do not have a restaurant with that id'
+    }, status=status.HTTP_400_BAD_REQUEST)
+
+
+@api_view(['GET'])
+@permission_classes([IsAuthenticated])
+def get_tickets(request, restaurant_id):
+  try:
+    user = get_profile_from_api_request(request=request)
+    tickets = Ticket.objects.filter(
+        restaurant__id=restaurant_id,
+        is_deleted=False,
+        restaurant__owner__id=user.id,
+    )
+    serializer = TicketSerializer(tickets, many=True)
+
+    return Response(data=serializer.data, status=status.HTTP_200_OK)
+  except ValidationError:
+    return Response({
+        'error': 'Invalid restaurant id'
+    }, status=status.HTTP_400_BAD_REQUEST)
+
+
+@api_view(['GET'])
+@permission_classes([AllowAny])
+def get_single_ticket(request, ticket_id):
+  try:
+    ticket = Ticket.objects.get(id=ticket_id, is_deleted=False)
+    serializer = TicketSerializer(ticket, many=False)
+
+    return Response(data=serializer.data, status=status.HTTP_200_OK)
+  except (ValidationError, ObjectDoesNotExist):
+    return Response({
+        'error': 'Invalid ticket id'
+    }, status=status.HTTP_400_BAD_REQUEST)
+
+
+@api_view(['PUT'])
+@permission_classes([IsAuthenticated])
+def update_ticket(request, ticket_id):
+  data = request.data
+  user = get_profile_from_api_request(request=request)
+
+  try:
+    ticket = Ticket.objects.get(id=ticket_id, restaurant__owner__id=user.id)
+    ticket_serializer = TicketSerializer(ticket)
+    ticket_data = ticket_serializer.data
+    ticket_data.pop('restaurant', None)
+    ticket_data.pop('id', None)
+    ticket_data.update(request.data)
+
+    serializer = TicketUpdateSerializer(data=ticket_data)
+    serializer.is_valid(raise_exception=True)
+    Ticket.objects.update_or_create(
+        id=ticket_id,
+        defaults=serializer.data
+    )
+
+    return Response(data=serializer.data, status=status.HTTP_200_OK)
+  except (ObjectDoesNotExist, ValidationError):
+    return Response({
+        'error': 'invalid ticket id'
+    }, status=status.HTTP_400_BAD_REQUEST)
+
+
+@api_view(['PUT'])
+@permission_classes([AllowAny])
+def buy_ticket(request, ticket_id):
```

Comment:
please try to insert this method into the viewset

Reply:
> please try to insert this method into the viewset

Hello @chaodev15,

I have implemented that, kindly check it out.

Thanks, learning a lot and embracing the beauty of class based views